### PR TITLE
Sync with minimal repo setup guide in .github

### PR DESCRIPTION
**Pull request by tovrstra**
This brings the IOData repo in line with the instructions in https://github.com/theochem/.github/pull/7

See #313 and https://github.com/theochem/.github/issues/5 for more background.

I've added a relatively conservative max file size, but we can still make exceptions if needed. IOData does not use Git LFS, which I'm inclined to keep that way. My experiences so far with Git LFS have not been great.

## Summary by Sourcery

This pull request updates the .pre-commit-config.yaml file to align with the minimal repository setup guide, including setting a conservative max file size limit and excluding binary files from CRLF removal.

- **Chores**:
    - Updated .pre-commit-config.yaml to align with the minimal repo setup guide, including setting a max file size limit of 100KB for added files and excluding binary files from CRLF removal.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -26,12 +27,13 @@
   rev: v1.5.5
   hooks:
     - id: remove-crlf
+      exclude_types: [binary]
```
[the remark this answers is not in the document]
Comment (tovrstra) [reply .pre-commit-config.yaml:30]: too obvious

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -4,15 +4,16 @@ repos:
   rev: v4.6.0
   hooks:
     - id: check-added-large-files
+      args: ["--maxkb=100"]
```
[the remark this answers is not in the document]
Comment (tovrstra) [reply .pre-commit-config.yaml:7]: I'll first try this out. It's a nice idea.

[the remark this answers is not in the document]
Comment (tovrstra) [reply .pre-commit-config.yaml:7]: Does not work.

**Review by PaulWAyers — APPROVED** [review]
LGTM

**Comment by tovrstra** [comment]
Thanks for reviewing. Merging...

**Pull request merged**